software/01.React/README.md

```diff
@@ -1,241 +1,122 @@
---header 'Content-Type: application/json' \
---data-raw '{
-	"name": "YOUR_FAKE_NAME",
-	"email": "YOUR_FAKE_EMAIL",
-	"password": "YOUR_LONG_PASSWORD",
-	"age": YOUR_FAKE_AGE
-}'
-```
+From now, you are able to create static components that don't interact with the user.
+We will now create dynamic components to allow the user to add a new task to the list, using [React hooks](https://reactjs.org/docs/hooks-intro.html).
 
-This will return you something like:
-
-```json
-{
-   "user":{
-      "age":YOUR_FAKE_AGE,
-      "_id":"5f61fb805ac0a50017bd8867",
-      "name":"YOUR_FAKE_NAME",
-      "email":"YOUR_FAKE_EMAIL",
-      "createdAt":"2020-09-16T11:48:16.439Z",
-      "updatedAt":"2020-09-16T11:48:16.789Z",
-      "__v":1
-   },
-   "token":"eyJhbGgiOiJIUzI1NiIsInR5cCI6IkpXVCJ9.ayJfaWQiOiI2ZjYxZmI4MDVpYzBhNTAwMTdiZPg4NjciLCJpYXQiOjE2MDAyNTY4OTZ9.2skAVYPi4vS2bDd0dQN9Melu9aqNY-13APzyqyQ9Q-4"
-}
-```
-
-The `token` field contains the string that you will use to be identified. By adding it in your requests, the API will know who you are and which task you have!
+To create a task, we need to create forms where the user can enter a title, a description and a due date with a button to validate. Let's create them with [Material UI](https://mui.com/components/)!
 
-If you want to add a new note, just do:
-
-```sh
-curl --location --request POST 'https://api-nodejs-todolist.herokuapp.com/task' \
---header 'Authorization: Bearer REPLACE_YOUR_TOKEN_HERE' \
---header 'Content-Type: application/json' \
---data-raw '{
-	"description": "finish workshop"
-}'
-```
-
-You should receive a JSON object saying `succes: true` with the note you just added.
-
-Finally, if you want to see all your notes, just do:
-
-```sh
-curl --location --request GET 'https://api-nodejs-todolist.herokuapp.com/task' \
---header 'Authorization: Bearer REPLACE_YOUR_TOKEN_HERE' \
---header 'Content-Type: application/json'
-```
+Let's then use the [useState](https://www.freecodecamp.org/news/introduction-to-react-hooks/) hook to handle user input and button clicks!
 
-Of course, there is a request to delete a note, but find it by yourself on the documentation, it should not be very hard!
+:bulb: Mix the usage of `useState` with `props`!
 
-### Use the API in React
+> We recommend you to create a small '+' button in one of the corner of your app that will open a small window containing the title and description forms as well as the button to add the task.
 
-You've probably saw it in the documentation, but examples in other languages are available. The one we want is JavaScript Fetch because we will use it directly in our react code. What you will have to do now is to call theses functions to send the result in your `Task` component. First, create a `request.js` file in the `src` folder containing this code:
+:bulb: You can customize your hooks to avoid unwanted cases, such as an empty field.
+<br/><br/>
+## Step 5: Discover another hook: React contexts
```

Comment:
React context is a complicated part of React, you need to give more explanations.